e2e_tests/tests/cluster/test_master_restart.py

```diff
@@ -147,12 +147,12 @@ def test_master_restart_error_missing_docker_container(
 
         expected_message = (
             (
-                "allocation failed due to agent failure: agent failed while the "
+                "allocation terminated after agent failed while the "
```

Comment:
There seems to be some difference in behavior here triggered by my changes -- I don't know if its significant enough to require a fix outside of e2e or if we're okay with this:
In `master/internal/task/allocation.go`:
```
func (a *Allocation) terminated(ctx *actor.Context, reason string) {
	...
	exitReason := fmt.Sprintf("allocation terminated after %s", reason)
	switch {
	...
		case sproto.ResourcesFailure:
			switch err.FailureType {
			...
			case sproto.AgentError, sproto.AgentFailed:
				exitReason = fmt.Sprintf("allocation failed due to agent failure: %s", err)
				ctx.Log().Warn(exitReason)
				...
			case sproto.RestoreError:
				exitReason = fmt.Sprintf("allocation failed due to restore error: %s", err)
				ctx.Log().Warn(exitReason)
				...
```
Tricky part is that I don't see where `exitReason` is being returned _before_ the appropriate cases are triggered.

Reply:
hmm if you're unsure just avoid making this change if it isn't a direct goal of the pr to keep it simpler?